.gitlab/test_python.sh

```
@@ -93,19 +93,48 @@ python3 partial_md_example.py --init-port "$(get_next_tcp_port)" --target-port "
+# Run a two-peers example: starts a target on an OS-assigned port,
+# reads the port from the target's NIXL_INFO log line on stderr,
+# then launches the initiator against it.
+# Extra arguments are passed as env vars to the initiator.
+# Usage: run_two_peers <script> [ENV=val ...]
+run_two_peers() {
+    local script=$1
+    shift
+
+    local target_log
+    target_log=$(mktemp)
+    trap "rm -f '$target_log'" EXIT
```

Shellcheck (0.11.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)